core/src/main/java/org/eclipse/hono/connection/impl/ConnectionFactoryImpl.java

```diff
@@ -44,6 +45,7 @@
 public final class ConnectionFactoryImpl implements ConnectionFactory {
 
     private static final Logger logger = LoggerFactory.getLogger(ConnectionFactoryImpl.class);
+    private final String containerIdUuidPart = UUID.randomUUID().toString();
```

Comment:
this means that all connections produced by this factory share the same UUID, right? Doesn't it make more sense that each connection has a distinct UUID?

Reply:
In practise I guess we mostly use one factory instance only for one `HonoConnection` instance, that would mean a `HonoConnection` instance gets a distinct UUID. But you are right, it would make more sense to enforce this by keeping the UUID in the `HonoConnection`.
I don't think we need a distinct UUID per `ProtonConnection` - see my other comment.